### PR TITLE
add stompy mini

### PR DESCRIPTION
**Pull request by budzianowski**

### CODE REVIEW

**Review by budzianowski — COMMENTED** [review]
Nice! Can we add this as a separate file so we can run two different embodiments?

**Comment by WT-MM** [comment]
> Nice! Can we add this as a separate file so we can run two different embodiments?

Sure - was actually thinking about how we might want to add a config for this so that it'd be easier/clearer to swap out embodiments

**Review by WT-MM — COMMENTED** [review]
Looks good to me! Great job!

File: demo.py [hunk 1]
```diff
@@ -4,123 +4,146 @@
...
+URDF_WEB = "https://raw.githubusercontent.com/kscalelabs/teleop/pawel/add_stompy_mini/urdf/stompy_mini/upper_half_assembly_simplified.urdf"
+URDF_LOCAL = "urdf/stompy_mini/upper_half_assembly_simplified.urdf"
 UPDATE_RATE = 1
 
-# Robot configuration
-START_POS_TRUNK_PYBULLET: NDArray = np.array([0, 0, 1])
-START_EUL_TRUNK_PYBULLET: NDArray = np.array([-math.pi / 2, 0, 2.15])
-START_POS_TRUNK_VUER: NDArray = np.array([0, 1, 0])
-START_EUL_TRUNK_VUER: NDArray = np.array([-math.pi, -0.68, 0])
+# # Robot configuration
```
Comment (budzianowski) [inline demo.py:29]: Do we keep it or remove it?

File: demo.py [hunk 2]
```diff
@@ -4,123 +4,146 @@
...
-    "joint_full_arm_5_dof_1_lower_arm_1_dof_1_hand_1_slider_2",
-]
-EER_CHAIN_HAND = [
-    "joint_full_arm_5_dof_2_lower_arm_1_dof_1_hand_1_slider_1",
-    "joint_full_arm_5_dof_2_lower_arm_1_dof_1_hand_1_slider_2",
-]
-
-OFFSET = list(START_Q.values())
-OFFSET_LEFT = [START_Q[joint] for joint in EEL_CHAIN_ARM + EEL_CHAIN_HAND]
+# # Kinematic chains for each arm
```
Comment (budzianowski) [inline demo.py:66]: Same here?

File: demo.py [hunk 3]
```diff
@@ -4,123 +4,146 @@
...
 # Hand tracking parameters
 INDEX_FINGER_TIP_ID, THUMB_FINGER_TIP_ID, MIDDLE_FINGER_TIP_ID = 8, 4, 14
 PINCH_DIST_CLOSED, PINCH_DIST_OPENED = 0.1, 0.1  # 10 cm
 EE_S_MIN, EE_S_MAX = 0.0, 0.05
 
 # Global variables
-q_lock = asyncio.Lock()
-q = deepcopy(START_Q)
-goal_pos_eel, goal_pos_eer = START_POS_EEL, START_POS_EER
+# q_lock = asyncio.Lock()
```
Comment (budzianowski) [inline demo.py:95]: remove?

File: demo.py [hunk 4]
```diff
@@ -4,123 +4,146 @@
...
+            np.array(self.config["start_pos_eer"]),
+        )
+        self.goal_pos_eel, self.goal_pos_eer = (
+            np.array(self.config["start_pos_eel"]),
+            np.array(self.config["start_pos_eer"]),
+        )
+        self.q = deepcopy(self.config["start_q"])
         self.q_lock = asyncio.Lock()
 
+        # Offset between Vuer and PyBullet trunk positions / coordinates
```
Comment (budzianowski) [inline demo.py:125]: Does it apply to the full stompy too?

File: demo.py [hunk 5]
```diff
@@ -200,8 +228,8 @@ async def inverse_kinematics(self, arm: str, max_attempts: int = 20) -> float |
             lowerLimits=lower_limits,
             upperLimits=upper_limits,
             jointRanges=joint_ranges,
-            restPoses=OFFSET,
-            jointDamping=joint_damping,
+            restPoses=self.offsets["offset_left" if arm == "left" else "offset_right"],
+            # jointDamping=joint_damping,
```
Comment (budzianowski) [inline demo.py:232]: remove?

File: config.yaml [hunk 6]
```diff
@@ -1,48 +1,76 @@
...
-    motor_config:
-      arm:
-        signs: [1, -1, 1, 1, 1, -1]
-        # signs: [1, 1, 1, -1, 1, -1]
-        increments: [4, 4, 4, 4, 4, 4]
-        maximum_values: [80, 80, 80, 70, 0, 10]
-        offsets: [0, 0, 0, 0, 0, 0]
+embodiments:
+  stompy_mini:
+    sides: ["right"]
```
Comment (budzianowski) [inline config.yaml:3]: what is a side?

File: robot_config.yaml [hunk 7]
```diff
@@ -0,0 +1,101 @@
+robots:
```
Comment (budzianowski) [inline robot_config.yaml:1]: that should be on the firmware side

**Review by budzianowski — COMMENTED** [review]
can you remove unused urdfs?

File: config.yaml (hunk 6 above)
Comment (WT-MM) [reply config.yaml:3]: for teleop control - whether we want to use left or right hands (or both)

File: demo.py (hunk 4 above)
Comment (WT-MM) [reply demo.py:125]: yep, but it's 0 in the full stompy config

**Pull request merged**